### PR TITLE
improve validators based on expected type

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: apypie/action.py [hunk 1]
```diff
@@ -76,6 +81,10 @@ def _validate(self, params, values, path=None):
                             self._validate(param_description.params, item, self._add_to_path(path, param_description.name, num))
                     if param_description.expected_type == 'hash':
```
Comment (ekohl) [inline apypie/action.py]: Unrelated

```suggestion
                    elif param_description.expected_type == 'hash':
```

**Comment by evgeni** [comment]
Yeah, I had them in separate if branches before and it was also ugly.
